src/managers/ThreadManager.js

```diff
@@ -0,0 +1,231 @@
+   * @property {boolean} [fetchAll=false] When type is `private` whether to fetch **all** archived threads,
+   * requires `MANAGE_THREADS` if true
+   * @property {DateResolvable|ThreadChannelResolvable} [before] Identifier for a Date or Snowflake
+   * to get threads that were created before it,
+   * must be a ThreadChannelResolvable when type is `private` and fetchAll is `false`
+   * @property {number} [limit] Maximum number of threads to return
+   */
+
+  /**
+   * The data returned from a thread fetch that returns multiple threads.
+   * @typedef {FetchedThreads}
+   * @property {Collection<Snowflake, ThreadChannel>} threads The threads fetched, with any members returned
+   * @property {?boolean} hasMore Whether there are potentially additional threads that require a subsequent call
+   */
+
+  /**
+   * Obtains a set of archived threads from Discord, requires `READ_MESSAGE_HISTORY` in the parent channel.
+   * @param {FetchArchivedThreadOptions} [options] The options to use when fetch archived threads
+   * @param {boolean} [cache=true] Whether to cache the new thread objects if they aren't already
+   * @returns {Promise<FetchedThreads>}
+   */
+  async fetchArchived({ type = 'public', fetchAll = false, before, limit } = {}, cache = true) {
+    let path = this.client.api.channels(this.channel.id);
+    if (type === 'private' && fetchAll) {
+      path = path.users('@me');
+    }
+    let timestamp;
+    let id;
+    if (typeof before !== 'undefined') {
+      if (before instanceof ThreadChannel || /^\d{16,19}$/.test(String(before))) {
+        id = this.resolveID(before);
+        timestamp = this.resolve(before)?.archivedAt?.toISOString();
+      } else {
+        try {
+          timestamp = new Date(before).toISOString();
+        } catch {
+          throw new TypeError('INVALID_TYPE', 'before', 'DateResolvable or ThreadChannelResolvable');
+        }
+      }
+    }
+    const raw = await path.threads
+      .archived(type)
+      .get({ query: { before: type === 'private' && !fetchAll ? id : timestamp, limit } });
+    return this._mapThreads(raw, cache);
+  }
+
+  /**
+   * Obtains the accessible active threads from Discord, requires `READ_MESSAGE_HISTORY` in the parent channel.
+   * @param {boolean} [cache=true] Whether to cache the new thread objects if they aren't already
+   * @returns {Promise<FetchedThreads>}
+   */
+  async fetchActive(cache = true) {
+    const raw = await this.client.api.channels(this.channel.id).threads.active.get();
+    return this._mapThreads(raw, cache);
+  }
+
+  _mapThreads(rawThreads, cache) {
+    const threads = rawThreads.threads.reduce((col, raw) => {
+      const thread = this.client.channels.add(raw, null, cache);
+      return col.set(thread.id, thread);
+    }, new Collection());
+    rawThreads.members.forEach(rawMember => this.client.channels.cache.get(rawMember.id)?.members.add(rawMember));
```

Comment:
Can't you tap into the `threads` collection you just created, since that ensures you have the thread made?

Reply:
nice call